src/replication/DealReplicationWorker.ts

```diff
@@ -289,15 +295,15 @@ export default class DealReplicationWorker extends BaseService {
     }
     try {
       const providers = DealReplicationWorker.generateProvidersList(replicationRequest.storageProviders);
-      // Find cars that are finished generation
-      const cars = await Datastore.GenerationRequestModel.find({
-        datasetId: replicationRequest.datasetId,
-        status: 'completed'
-      })
-        .sort({
-          pieceCid: 1
-        });
       const makeDealAll = providers.map(async (provider) => {
+        // Find cars that are finished generation
```

Comment:
why is it needed to move inside the closure?

Reply:
I'm suspecting the commP mysterious changed bug is due to racing conditions and I want to move this inside closure to avoid side-effect.